### PR TITLE
Http cache

### PR DESCRIPTION
**Pull request by mnot**
Add tests for HTTP caching in Fetch. See also <https://github.com/whatwg/fetch/pull/496>.

Caveats in README.md.

### CODE REVIEW

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5137/)

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/210877075)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 9ae2cd6729d65deda8642011067124fb1f44e415
Using browser at version BuildID 20170313110417; SourceStamp f9362554866b327700c7f9b18050d7b7eb3d2b23
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>8 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5137/fetch/http-cache/304-update.html">/fetch/http-cache/304-update.html</a></summary>


|                             Subtest                             | Results | Messages |
|-----------------------------------------------------------------|---------|----------|
|                                                                 | OK      |          |
| `HTTP cache updates returned headers from a Last-Modified 304.` | PASS    |          |
| `HTTP cache updates stored headers from a Last-Modified 304.`   | PASS    |          |
| `HTTP cache updates returned headers from a ETag 304.`          | PASS    |          |
| `HTTP cache updates stored headers from a ETag 304.`            | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5137/fetch/http-cache/cc-request.html">/fetch/http-cache/cc-request.html</a></summary>


|                                                                    Subtest                                                                     | Results |                  Messages                 |
|------------------------------------------------------------------------------------------------------------------------------------------------|---------|-------------------------------------------|
|                                                                                                                                                | OK      |                                           |
| `HTTP cache does use aged but fresh response when request contains Cache-Control: max-age that's is smaller than remaining freshness.`         | PASS    |                                           |
| `HTTP cache doesn't use aged but fresh response when request contains Cache-Control: max-age=0.`                                               | PASS    |                                           |
| `HTTP cache doesn't use aged but fresh response when request contains Cache-Control: max-age=1.`                                               | PASS    |                                           |
| `HTTP cache does use fresh response with Age header when request contains Cache-Control: max-age that is smaller than remaining freshness.`    | PASS    |                                           |
| `HTTP cache doesn't use fresh response with Age header when request contains Cache-Control: max-age that is greater than remaining freshness.` | PASS    |                                           |
| `HTTP cache does use aged stale response when request contains Cache-Control: max-stale that permits its use.`                                 | PASS    |                                           |
| `HTTP cache does reuse stale response with Age header when request contains Cache-Control: max-stale that permits its use.`                    | PASS    |                                           |
| `HTTP cache doesn't reuse fresh response when request contains Cache-Control: min-fresh that wants it fresher.`                                | PASS    |                                           |
| `HTTP cache doesn't reuse fresh response with Age header when request contains Cache-Control: min-fresh that wants it fresher.`                | PASS    |                                           |
| `HTTP cache doesn't reuse fresh response when request contains Cache-Control: no-cache.`                                                       | PASS    |                                           |
| `HTTP cache doesn't reuse fresh response when request contains Cache-Control: no-store.`                                                       | PASS    |                                           |
| `HTTP cache generates 504 status code when nothing is in cache and request contains Cache-Control: only-if-cached.`                            | FAIL    | `assert_equals: expected 504 but got 200` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5137/fetch/http-cache/freshness.html">/fetch/http-cache/freshness.html</a></summary>


|                                              Subtest                                               | Results |                           Messages                           |
|----------------------------------------------------------------------------------------------------|---------|--------------------------------------------------------------|
|                                                                                                    | OK      |                                                              |
| `HTTP cache reuses a response with a future Expires.`                                              | PASS    |                                                              |
| `HTTP cache does not reuse a response with a past Expires.`                                        | PASS    |                                                              |
| `HTTP cache does not reuse a response with a present Expires.`                                     | PASS    |                                                              |
| `HTTP cache does not reuse a response with an invalid Expires.`                                    | PASS    |                                                              |
| `HTTP cache reuses a response with positive Cache-Control: max-age.`                               | PASS    |                                                              |
| `HTTP cache does not reuse a response with Cache-Control: max-age=0.`                              | PASS    |                                                              |
| `HTTP cache reuses a response with positive Cache-Control: max-age and a past Expires.`            | PASS    |                                                              |
| `HTTP cache does not reuse a response with Cache-Control: max-age=0 and a future Expires.`         | PASS    |                                                              |
| `HTTP cache does not prefer Cache-Control: s-maxage over Cache-Control: max-age.`                  | PASS    |                                                              |
| `HTTP cache does not reuse a response when the Age header is greater than its freshness lifetime.` | FAIL    | `assert_false: Cached response used expected false got true` |
| `HTTP cache does not store a response with Cache-Control: no-store.`                               | PASS    |                                                              |
| `HTTP cache stores a response with Cache-Control: no-cache, but revalidates upon use.`             | PASS    |                                                              |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5137/fetch/http-cache/heuristic.html">/fetch/http-cache/heuristic.html</a></summary>


|                                                                   Subtest                                                                   | Results |                           Messages                          |
|---------------------------------------------------------------------------------------------------------------------------------------------|---------|-------------------------------------------------------------|
|                                                                                                                                             | OK      |                                                             |
| `HTTP cache reuses a 200 response with Last-Modified based upon heuristic freshness.`                                                       | PASS    |                                                             |
| `HTTP cache reuses an unknown response with Last-Modfied based upon heuristic freshness when Cache-Control: public is present.`             | FAIL    | `assert_true: Cached response used expected true got false` |
| `HTTP cache does not reuse an unknown response with Last-Modfied based upon heuristic freshness when Cache-Control: public is not present.` | PASS    |                                                             |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5137/fetch/http-cache/invalidate.html">/fetch/http-cache/invalidate.html</a></summary>


|                                               Subtest                                                | Results |                           Messages                          |
|------------------------------------------------------------------------------------------------------|---------|-------------------------------------------------------------|
|                                                                                                      | OK      |                                                             |
| `HTTP cache invalidates after a successful response from a POST`                                     | PASS    |                                                             |
| `HTTP cache does not invalidate after a failed response from an unsafe request`                      | FAIL    | `assert_true: Cached response used expected true got false` |
| `HTTP cache invalidates after a successful response from a PUT`                                      | PASS    |                                                             |
| `HTTP cache invalidates after a successful response from a DELETE`                                   | PASS    |                                                             |
| `HTTP cache invalidates after a successful response from an unknown method`                          | PASS    |                                                             |
| `HTTP cache invalidates Location URL after a successful response from a POST`                        | PASS    |                                                             |
| `HTTP cache does not invalidate Location URL after a failed response from an unsafe request`         | FAIL    | `assert_true: Cached response used expected true got false` |
| `HTTP cache invalidates Location URL after a successful response from a PUT`                         | PASS    |                                                             |
| `HTTP cache invalidates Location URL after a successful response from a DELETE`                      | PASS    |                                                             |
| `HTTP cache invalidates Location URL after a successful response from an unknown method`             | PASS    |                                                             |
| `HTTP cache invalidates Content-Location URL after a successful response from a POST`                | PASS    |                                                             |
| `HTTP cache does not invalidate Content-Location URL after a failed response from an unsafe request` | FAIL    | `assert_true: Cached response used expected true got false` |
| `HTTP cache invalidates Content-Location URL after a successful response from a PUT`                 | PASS    |                                                             |
| `HTTP cache invalidates Content-Location URL after a successful response from a DELETE`              | PASS    |                                                             |
| `HTTP cache invalidates Content-Location URL after a successful response from an unknown method`     | PASS    |                                                             |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5137/fetch/http-cache/partial.html">/fetch/http-cache/partial.html</a></summary>


|                                 Subtest                                  | Results |                                   Messages                                  |
|--------------------------------------------------------------------------|---------|-----------------------------------------------------------------------------|
|                                                                          | OK      |                                                                             |
| `HTTP cache stores partial content and reuses it.`                       | FAIL    | `assert_true: Cached response used expected true got false`                 |
| `HTTP cache stores complete response and serves smaller ranges from it.` | FAIL    | `assert_true: Cached response used expected true got false`                 |
| `HTTP cache stores partial response and serves smaller ranges from it.`  | FAIL    | `assert_true: Cached response used expected true got false`                 |
| `HTTP cache stores partial content and completes it.`                    | FAIL    | `assert_equals: expected (string) "bytes=5-" but got (undefined) undefined` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5137/fetch/http-cache/status.html">/fetch/http-cache/status.html</a></summary>


|                                 Subtest                                  | Results |                           Messages                          |
|--------------------------------------------------------------------------|---------|-------------------------------------------------------------|
|                                                                          | OK      |                                                             |
| `HTTP cache goes to the network if it has a stale 200 response.`         | PASS    |                                                             |
| `HTTP cache avoids going to the network if it has a fresh 200 response.` | PASS    |                                                             |
| `HTTP cache goes to the network if it has a stale 203 response.`         | PASS    |                                                             |
| `HTTP cache avoids going to the network if it has a fresh 203 response.` | PASS    |                                                             |
| `HTTP cache goes to the network if it has a stale 204 response.`         | PASS    |                                                             |
| `HTTP cache avoids going to the network if it has a fresh 204 response.` | FAIL    | `assert_true: Cached response used expected true got false` |
| `HTTP cache goes to the network if it has a stale 299 response.`         | PASS    |                                                             |
| `HTTP cache avoids going to the network if it has a fresh 299 response.` | FAIL    | `assert_true: Cached response used expected true got false` |
| `HTTP cache goes to the network if it has a stale 400 response.`         | PASS    |                                                             |
| `HTTP cache avoids going to the network if it has a fresh 400 response.` | FAIL    | `assert_true: Cached response used expected true got false` |
| `HTTP cache goes to the network if it has a stale 404 response.`         | PASS    |                                                             |
| `HTTP cache avoids going to the network if it has a fresh 404 response.` | FAIL    | `assert_true: Cached response used expected true got false` |
| `HTTP cache goes to the network if it has a stale 410 response.`         | PASS    |                                                             |
| `HTTP cache avoids going to the network if it has a fresh 410 response.` | PASS    |                                                             |
| `HTTP cache goes to the network if it has a stale 499 response.`         | PASS    |                                                             |
| `HTTP cache avoids going to the network if it has a fresh 499 response.` | FAIL    | `assert_true: Cached response used expected true got false` |
| `HTTP cache goes to the network if it has a stale 500 response.`         | PASS    |                                                             |
| `HTTP cache avoids going to the network if it has a fresh 500 response.` | FAIL    | `assert_true: Cached response used expected true got false` |
| `HTTP cache goes to the network if it has a stale 502 response.`         | PASS    |                                                             |
| `HTTP cache avoids going to the network if it has a fresh 502 response.` | FAIL    | `assert_true: Cached response used expected true got false` |
| `HTTP cache goes to the network if it has a stale 503 response.`         | PASS    |                                                             |
| `HTTP cache avoids going to the network if it has a fresh 503 response.` | FAIL    | `assert_true: Cached response used expected true got false` |
| `HTTP cache goes to the network if it has a stale 504 response.`         | PASS    |                                                             |
| `HTTP cache avoids going to the network if it has a fresh 504 response.` | FAIL    | `assert_true: Cached response used expected true got false` |
| `HTTP cache goes to the network if it has a stale 599 response.`         | PASS    |                                                             |
| `HTTP cache avoids going to the network if it has a fresh 599 response.` | FAIL    | `assert_true: Cached response used expected true got false` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5137/fetch/http-cache/vary.html">/fetch/http-cache/vary.html</a></summary>


|                                       Subtest                                       | Results | Messages |
|-------------------------------------------------------------------------------------|---------|----------|
|                                                                                     | OK      |          |
| `HTTP cache reuses Vary response when request matches.`                             | PASS    |          |
| `HTTP cache doesn't use Vary response when request doesn't match.`                  | PASS    |          |
| `HTTP cache doesn't use Vary response when request omits variant header.`           | PASS    |          |
| `HTTP cache reuses two-way Vary response when request matches.`                     | PASS    |          |
| `HTTP cache doesn't use two-way Vary response when request doesn't match.`          | PASS    |          |
| `HTTP cache doesn't use two-way Vary response when request omits variant header.`   | PASS    |          |
| `HTTP cache reuses three-way Vary response when request matches.`                   | PASS    |          |
| `HTTP cache doesn't use three-way Vary response when request doesn't match.`        | PASS    |          |
| `HTTP cache doesn't use three-way Vary response when request omits variant header.` | PASS    |          |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/210877076)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 9ae2cd6729d65deda8642011067124fb1f44e415
Using browser at version 58.0.3029.14 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>8 tests ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5137/fetch/http-cache/304-update.html">/fetch/http-cache/304-update.html</a></summary>


|                             Subtest                             | Results | Messages |
|-----------------------------------------------------------------|---------|----------|
|                                                                 | OK      |          |
| `HTTP cache updates returned headers from a Last-Modified 304.` | PASS    |          |
| `HTTP cache updates stored headers from a Last-Modified 304.`   | PASS    |          |
| `HTTP cache updates returned headers from a ETag 304.`          | PASS    |          |
| `HTTP cache updates stored headers from a ETag 304.`            | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5137/fetch/http-cache/cc-request.html">/fetch/http-cache/cc-request.html</a></summary>


|                                                                    Subtest                                                                     | Results |                           Messages                           |
|------------------------------------------------------------------------------------------------------------------------------------------------|---------|--------------------------------------------------------------|
|                                                                                                                                                | OK      |                                                              |
| `HTTP cache does use aged but fresh response when request contains Cache-Control: max-age that's is smaller than remaining freshness.`         | PASS    |                                                              |
| `HTTP cache doesn't use aged but fresh response when request contains Cache-Control: max-age=0.`                                               | PASS    |                                                              |
| `HTTP cache doesn't use aged but fresh response when request contains Cache-Control: max-age=1.`                                               | FAIL    | `assert_false: Cached response used expected false got true` |
| `HTTP cache does use fresh response with Age header when request contains Cache-Control: max-age that is smaller than remaining freshness.`    | PASS    |                                                              |
| `HTTP cache doesn't use fresh response with Age header when request contains Cache-Control: max-age that is greater than remaining freshness.` | FAIL    | `assert_false: Cached response used expected false got true` |
| `HTTP cache does use aged stale response when request contains Cache-Control: max-stale that permits its use.`                                 | FAIL    | `assert_true: Cached response used expected true got false`  |
| `HTTP cache does reuse stale response with Age header when request contains Cache-Control: max-stale that permits its use.`                    | FAIL    | `assert_true: Cached response used expected true got false`  |
| `HTTP cache doesn't reuse fresh response when request contains Cache-Control: min-fresh that wants it fresher.`                                | FAIL    | `assert_false: Cached response used expected false got true` |
| `HTTP cache doesn't reuse fresh response with Age header when request contains Cache-Control: min-fresh that wants it fresher.`                | FAIL    | `assert_false: Cached response used expected false got true` |
| `HTTP cache doesn't reuse fresh response when request contains Cache-Control: no-cache.`                                                       | PASS    |                                                              |
| `HTTP cache doesn't reuse fresh response when request contains Cache-Control: no-store.`                                                       | FAIL    | `assert_false: Cached response used expected false got true` |
| `HTTP cache generates 504 status code when nothing is in cache and request contains Cache-Control: only-if-cached.`                            | FAIL    | `assert_equals: expected 504 but got 200`                    |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5137/fetch/http-cache/freshness.html">/fetch/http-cache/freshness.html</a></summary>


|                                              Subtest                                               | Results | Messages |
|----------------------------------------------------------------------------------------------------|---------|----------|
|                                                                                                    | OK      |          |
| `HTTP cache reuses a response with a future Expires.`                                              | PASS    |          |
| `HTTP cache does not reuse a response with a past Expires.`                                        | PASS    |          |
| `HTTP cache does not reuse a response with a present Expires.`                                     | PASS    |          |
| `HTTP cache does not reuse a response with an invalid Expires.`                                    | PASS    |          |
| `HTTP cache reuses a response with positive Cache-Control: max-age.`                               | PASS    |          |
| `HTTP cache does not reuse a response with Cache-Control: max-age=0.`                              | PASS    |          |
| `HTTP cache reuses a response with positive Cache-Control: max-age and a past Expires.`            | PASS    |          |
| `HTTP cache does not reuse a response with Cache-Control: max-age=0 and a future Expires.`         | PASS    |          |
| `HTTP cache does not prefer Cache-Control: s-maxage over Cache-Control: max-age.`                  | PASS    |          |
| `HTTP cache does not reuse a response when the Age header is greater than its freshness lifetime.` | PASS    |          |
| `HTTP cache does not store a response with Cache-Control: no-store.`                               | PASS    |          |
| `HTTP cache stores a response with Cache-Control: no-cache, but revalidates upon use.`             | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5137/fetch/http-cache/heuristic.html">/fetch/http-cache/heuristic.html</a></summary>


|                                                                   Subtest                                                                   | Results |                           Messages                          |
|---------------------------------------------------------------------------------------------------------------------------------------------|---------|-------------------------------------------------------------|
|                                                                                                                                             | OK      |                                                             |
| `HTTP cache reuses a 200 response with Last-Modified based upon heuristic freshness.`                                                       | PASS    |                                                             |
| `HTTP cache reuses an unknown response with Last-Modfied based upon heuristic freshness when Cache-Control: public is present.`             | FAIL    | `assert_true: Cached response used expected true got false` |
| `HTTP cache does not reuse an unknown response with Last-Modfied based upon heuristic freshness when Cache-Control: public is not present.` | PASS    |                                                             |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5137/fetch/http-cache/invalidate.html">/fetch/http-cache/invalidate.html</a></summary>


|                                               Subtest                                                | Results |                           Messages                           |
|------------------------------------------------------------------------------------------------------|---------|--------------------------------------------------------------|
|                                                                                                      | OK      |                                                              |
| `HTTP cache invalidates after a successful response from a POST`                                     | PASS    |                                                              |
| `HTTP cache does not invalidate after a failed response from an unsafe request`                      | PASS    |                                                              |
| `HTTP cache invalidates after a successful response from a PUT`                                      | PASS    |                                                              |
| `HTTP cache invalidates after a successful response from a DELETE`                                   | PASS    |                                                              |
| `HTTP cache invalidates after a successful response from an unknown method`                          | FAIL    | `assert_false: Cached response used expected false got true` |
| `HTTP cache invalidates Location URL after a successful response from a POST`                        | FAIL    | `assert_false: Cached response used expected false got true` |
| `HTTP cache does not invalidate Location URL after a failed response from an unsafe request`         | PASS    |                                                              |
| `HTTP cache invalidates Location URL after a successful response from a PUT`                         | FAIL    | `assert_false: Cached response used expected false got true` |
| `HTTP cache invalidates Location URL after a successful response from a DELETE`                      | FAIL    | `assert_false: Cached response used expected false got true` |
| `HTTP cache invalidates Location URL after a successful response from an unknown method`             | FAIL    | `assert_false: Cached response used expected false got true` |
| `HTTP cache invalidates Content-Location URL after a successful response from a POST`                | FAIL    | `assert_false: Cached response used expected false got true` |
| `HTTP cache does not invalidate Content-Location URL after a failed response from an unsafe request` | PASS    |                                                              |
| `HTTP cache invalidates Content-Location URL after a successful response from a PUT`                 | FAIL    | `assert_false: Cached response used expected false got true` |
| `HTTP cache invalidates Content-Location URL after a successful response from a DELETE`              | FAIL    | `assert_false: Cached response used expected false got true` |
| `HTTP cache invalidates Content-Location URL after a successful response from an unknown method`     | FAIL    | `assert_false: Cached response used expected false got true` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5137/fetch/http-cache/partial.html">/fetch/http-cache/partial.html</a></summary>


|                                 Subtest                                  | Results |                                   Messages                                  |
|--------------------------------------------------------------------------|---------|-----------------------------------------------------------------------------|
|                                                                          | OK      |                                                                             |
| `HTTP cache stores partial content and reuses it.`                       | FAIL    | `assert_true: Cached response used expected true got false`                 |
| `HTTP cache stores complete response and serves smaller ranges from it.` | FAIL    | `assert_equals: expected 200 but got 206`                                   |
| `HTTP cache stores partial response and serves smaller ranges from it.`  | FAIL    | `assert_true: Cached response used expected true got false`                 |
| `HTTP cache stores partial content and completes it.`                    | FAIL    | `assert_equals: expected (string) "bytes=5-" but got (undefined) undefined` |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5137/fetch/http-cache/status.html">/fetch/http-cache/status.html</a></summary>


|                                 Subtest                                  | Results | Messages |
|--------------------------------------------------------------------------|---------|----------|
|                                                                          | OK      |          |
| `HTTP cache goes to the network if it has a stale 200 response.`         | PASS    |          |
| `HTTP cache avoids going to the network if it has a fresh 200 response.` | PASS    |          |
| `HTTP cache goes to the network if it has a stale 203 response.`         | PASS    |          |
| `HTTP cache avoids going to the network if it has a fresh 203 response.` | PASS    |          |
| `HTTP cache goes to the network if it has a stale 204 response.`         | PASS    |          |
| `HTTP cache avoids going to the network if it has a fresh 204 response.` | PASS    |          |
| `HTTP cache goes to the network if it has a stale 299 response.`         | PASS    |          |
| `HTTP cache avoids going to the network if it has a fresh 299 response.` | PASS    |          |
| `HTTP cache goes to the network if it has a stale 400 response.`         | PASS    |          |
| `HTTP cache avoids going to the network if it has a fresh 400 response.` | PASS    |          |
| `HTTP cache goes to the network if it has a stale 404 response.`         | PASS    |          |
| `HTTP cache avoids going to the network if it has a fresh 404 response.` | PASS    |          |
| `HTTP cache goes to the network if it has a stale 410 response.`         | PASS    |          |
| `HTTP cache avoids going to the network if it has a fresh 410 response.` | PASS    |          |
| `HTTP cache goes to the network if it has a stale 499 response.`         | PASS    |          |
| `HTTP cache avoids going to the network if it has a fresh 499 response.` | PASS    |          |
| `HTTP cache goes to the network if it has a stale 500 response.`         | PASS    |          |
| `HTTP cache avoids going to the network if it has a fresh 500 response.` | PASS    |          |
| `HTTP cache goes to the network if it has a stale 502 response.`         | PASS    |          |
| `HTTP cache avoids going to the network if it has a fresh 502 response.` | PASS    |          |
| `HTTP cache goes to the network if it has a stale 503 response.`         | PASS    |          |
| `HTTP cache avoids going to the network if it has a fresh 503 response.` | PASS    |          |
| `HTTP cache goes to the network if it has a stale 504 response.`         | PASS    |          |
| `HTTP cache avoids going to the network if it has a fresh 504 response.` | PASS    |          |
| `HTTP cache goes to the network if it has a stale 599 response.`         | PASS    |          |
| `HTTP cache avoids going to the network if it has a fresh 599 response.` | PASS    |          |

</details>
<details>
<summary><a href="http://w3c-test.org/submissions/5137/fetch/http-cache/vary.html">/fetch/http-cache/vary.html</a></summary>


|                                       Subtest                                       | Results | Messages |
|-------------------------------------------------------------------------------------|---------|----------|
|                                                                                     | OK      |          |
| `HTTP cache reuses Vary response when request matches.`                             | PASS    |          |
| `HTTP cache doesn't use Vary response when request doesn't match.`                  | PASS    |          |
| `HTTP cache doesn't use Vary response when request omits variant header.`           | PASS    |          |
| `HTTP cache reuses two-way Vary response when request matches.`                     | PASS    |          |
| `HTTP cache doesn't use two-way Vary response when request doesn't match.`          | PASS    |          |
| `HTTP cache doesn't use two-way Vary response when request omits variant header.`   | PASS    |          |
| `HTTP cache reuses three-way Vary response when request matches.`                   | PASS    |          |
| `HTTP cache doesn't use three-way Vary response when request doesn't match.`        | PASS    |          |
| `HTTP cache doesn't use three-way Vary response when request omits variant header.` | PASS    |          |

</details>
</details>

File: fetch/OWNERS [hunk 1]
```diff
@@ -1,3 +1,4 @@
 @jdm
 @youennf
 @annevk
+@mnot
```
Comment (annevk) [inline fetch/OWNERS]: Please add a trailing newline.

File: fetch/http-cache/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
+
```
Comment (annevk) [inline fetch/http-cache/README.md]: Remove leading newline.

**Comment by annevk** [comment]
These looks good to me, as does the change to the standard. Will you file browser bugs on failures or should I?

Also, you might want to consider using the [.any.js](http://web-platform-tests.org/writing-tests/testharness.html) convention to also test the HTTP cache from workers at close to no extra effort.

**Comment by mnot** [comment]
I'm happy to file bugs. Should I do that after the tests land, so I have something to point to (that's easier to run)?

WRT workers -- interesting. Do we think that the results will be any different? The cache behaviour *should* be the same.

**Comment by annevk** [comment]
Filing bugs after is fine. I think testing workers is good, since we often think they should behave the same, but then they don't because implementers took shortcuts because threading/processes is hard.

**Pull request merged**